lms/djangoapps/completion/api/v1/views.py

```diff
@@ -0,0 +1,80 @@
+""" API v1 views. """
+import logging
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import permissions
+
+from lms.djangoapps.completion.models import BlockCompletion
+from openedx.core.lib.api.permissions import IsStaffOrOwner
+
+log = logging.getLogger(__name__)
+
+
+class CompletionBatchView(APIView):
+    permission_classes = (permissions.IsAuthenticated, IsStaffOrOwner,)
+
+    def _validate(self, batch_object):
+        """
+        Performs validation on the batch object to make sure it is in the proper format.
+
+        Parameters:
+            batch_object: The object provided by the caller.
+
+        Raises:
+
+            KeyError:
+                If a key is not found in the object.
+
+            django.core.exceptions.ValidationError:
+                If a float is passed that is not between 0.0 and 1.0.
+        """
+        if 'user' not in batch_object:
+            raise KeyError("Key 'user' not found.")
+        if 'course_key' not in batch_object:
+            raise KeyError("Key 'course_key' not found.")
+        if 'blocks' not in batch_object:
+            raise KeyError("Key 'blocks' not found.")
+
+        return
+
+    def post(self, request, *args, **kwargs):
+        """
+        Inserts a batch of completions.
+
+        REST Endpoint Format:
+        {
+          "user": "username",
+          "course_key": "course-key",
+          "blocks": {
+            "block_key1": 0.0,
+            "block_key2": 1.0,
+            "block_key3": 1.0,
+          }
+        }
+
+        **Returns**
+
+        * 200 if successful.
+        * 400 if an invalid parameter is passed.
+        """
+        batch_object = request.data or {}
+        try:
+            self._validate(batch_object)
+        except KeyError:
+            return Response(400)
+        except ValidationError:
+            return Response(400)
+        user = batch_object['user']
+        course_key = batch_object['course_key']
+        blocks = batch_object['blocks']
+        user = User.objects.get(username=user)
+
+        try:
+            BlockCompletion.objects.submit_batch_completion(user, course_key, blocks)
+        except Exception:
```

Comment:
Catching generic `Exception`s is bad practice.. needs to be the specific Exception types that might be thrown.

Note that you can chain them, e.g.

```python
except (InvalidKeyError, ObjectDoesNotExist) as exc:
   do things
```